explorer/explorertypes.go

```diff
@@ -343,3 +325,74 @@ type NewMempoolTx struct {
 	Time int64
 	Hex  string
 }
+
+// ExtendedChainParams represents the data of ChainParams
+type ExtendedChainParams struct {
+	Params        *chaincfg.Params
+	AddressPrefix []AddrPrefix
+}
+
+// AddrPrefix represent the address name it's prefix and description
+type AddrPrefix struct {
+	Name        string
+	Prefix      string
+	Description string
+}
+
+// AddressPrefixes generates an array AddrPrefix by using chaincfg.Params
+func AddressPrefixes(params *chaincfg.Params) []AddrPrefix {
+	var Descriptions = []string{"First 2 letters of a P2PK address",
+		"First 2 letters of a P2PKH address",
+		"First 2 letters of an Edwards P2PKH address",
+		"First 2 letters of a secp256k1 Schnorr P2PKH address",
+		"First 2 letters of a P2SH address",
+		"First 2 letters of a WIF private key",
+		"BIP32 hierarchical deterministic extended key magics",
+		"BIP32 hierarchical deterministic extended key magics",
+	}
+
+	addrPrefix := make([]AddrPrefix, 0, len(Descriptions))
+
+	var Name = []string{"PubKeyAddrID",
+		"PubKeyHashAddrID",
+		"PKHEdwardsAddrID",
+		"PKHSchnorrAddrID",
+		"ScriptHashAddrID",
+		"PrivateKeyID",
+		"HDPrivateKeyID",
+		"HDPublicKeyID",
+	}
+	var MainnetPrefixes = []string{"Dk", "Ds", "De", "DS", "Dc", "Pm", "dprv", "dpub"}
+	var Testnet2Prefixes = []string{"Tk", "Ts", "Te", "TS", "Tc", "Pt", "tprv", "tpub"}
+	var SimnetPrefixes = []string{"Sk", "Ss", "Se", "SS", "Sc", "Ps", "sprv", "spub"}
+	name := params.Name
+	if name == "mainnet" {
+		for i, desc := range Descriptions {
+			addrPrefix = append(addrPrefix, AddrPrefix{
+				Name:        Name[i],
+				Description: desc,
+				Prefix:      MainnetPrefixes[i],
+			})
+		}
```

Comment:
Note that only `Prefix` is different, so please simplify:

```go
var netPrefixes []string
if name == "mainnet" {
    netPrefixes = MainnetPrefixes
} else if strings.HasPrefix(name, "testnet") {
    netPrefixes = TestnetPrefixes
} else if name == simnet" {
    netPrefixes = SimnetPrefixes
} else {
    return nil
}
```

so that you just have one loop:
```go
for i := range Descriptions {
	addrPrefix = append(addrPrefix, AddrPrefix{
		Name:        Name[i],
		Description: Descriptions[i],
		Prefix:      netPrefixes[i],
	})
}
```